packages/cli/src/deployment.ts

```diff
@@ -264,7 +264,10 @@ async function createDeployer<Settings = unknown>(
     console.log(`Deploying contract ${taskId}`)
     console.log(`Deployer - group ${signer.group} - ${signer.address}`)
     const result = await contract.transactionForDeployment(signer, params)
-    await signer.submitTransaction(result.unsignedTx)
+    await signer.signAndSubmitUnsignedTx({
+      unsignedTx: result.unsignedTx,
+      signerAddress: (await signer.getSelectedAccount()).address
```

Comment:
A `signerAddress` is required for each transaction because the selected account might have changed since the unsigned tx is built.

Signer will check the signer address before singing, this could provide more security. WDYT?